rfd/0xxx-tls-routing-behind-lb.md

```diff
@@ -0,0 +1,319 @@
+config yaml. This new config has no effect on how Proxy server serve incoming traffic but it will provide a hint to all
+the clients through the webapi ping:
+
+```
+{
+  "auth": {
+  ...
+  },
+  "proxy": {
+  ...
+    "tls_routing_enabled": true,
+    "tls_routing_upgrade_mode": "auto"
+  },
+  ...
+}
+```
+
+`auto` is the default mode where the client performances a test to decide if a connection upgrade is necessary. The test
+is done by sending a TLS handshake with a Teleport custom ALPN to the Proxy server:
+```
+                              ┌───────┐
+                         "on" │upgrade│ "off"
+ ┌────────────────────────────┤ mode  ├────────────────────────────────────┐
+ │                            └──┬────┘                                    │
+ │                               │ "auto" (default)                        │
+ │                               │                                         │
+ │                    ┌──────────▼────────────────┐                        │
+ │                    │     TLS Handshake         │                        │
+ │                    │ALPN teleport-reversetunnel│                        │
+ │                    └────┬─────────────────┬────┘                        │
+ │                         │                 │                             │
+ │                     ┌───▼─────┐        ┌──▼──────┐                      │
+ │                     │Handshake│        │Handshake├───────────────┐      │
+ │ ┌───────────────────┤ Success │        │ Failed  │               │      │
+ │ │                   └───┬─────┘        └┬────────┘               │      │
+ │ │                       │               │                        │      │
+ │ │Negotiated             │No ALPN        │remote error: tls:      │other │
+ │ │teleport-reversetunnel │Negotiated     │no application protocol │errors│
+ │ │                       │               │                        │      │
+ │ │                   ┌───▼────┐          │                  ┌─────▼──┐   │
+ │ │                   │Upgrade ◄──────────┘                  │  Not   │   │
+ └─┼───────────────────►Required│                             │Required◄───┘
+   │                   └────────┘                             └────▲───┘
+   │                                                               │
+   └───────────────────────────────────────────────────────────────┘
+```
+Connection upgrade for TLS Routing should be required when the client and the server fail to neogiate the ALPN protocol,
+hinting there is a load balancer in the middle that terminates TLS.
+
+However, this test may not be bullet proof. In particular, the test explicitly looks for `tls: no application protocol`
+from the remote when handshakes fails, but it is possible that a load balancer implementation decides to use a different
+text.
+
+In such cases, setting the mode to `on` can enforce the clients to always do the connection upgrade. Also, setting the
```

Comment:
Will this setting also be a part of cluster networking config resource? If so, let's give an example, if not and we don't want to allow to change it dynamically, let's mention it too.